### ISSUE TITLE
no error when duplicated rules in config file

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#12774]
```json
{
  "lint": {
    "select": [
      "RUF031",
      "RUF031"
    ],
    "ignore": [
      "D106",
      "D106"
    ]
  }
}
```

ruff should report an error when duplicated rules are present in the `select`/`ignore` lists, as it's an obvious mistake.

it could also account for redundant rules, eg:
```jsonc
{
  "lint": {
    "select": [
      "ALL",
      "D106"  // error: this rule is already covered by ALL
    ]
  }
}
```

**Comment by DetachHead** [comment]
also:

```json
{
  "lint": {
    "select": [
      "D106",
    ],
    "ignore": [
      "D106" // error: conflicts with select
    ]
  }
}
```

**Comment by MichaReiser** [comment]
I'm not sure if ruff should error in this case because it can do a decent job at recovering (it very much seems that you wanted to enable the rule and the ordering in `select` doesn't matter). 

I think what would be nice is if we can lint your pyproject.toml and point out that you repeated a value in the selector. I consider this similar to a lint rule that detects `set(True, True)` and proposes to simplify it to `set(True)`
